doc/sphinx/grammar.rst

```diff
@@ -0,0 +1,2247 @@
+              : `pattern99`
+   pattern200_list : `pattern200_list` | `pattern200`
+                   : `pattern200`
+   pattern99 : `pattern90`
+   pattern90 : `pattern10`
+   pattern10 : `pattern1` as `name`
+             : `pattern1` `pattern1_list`
+             : @ `reference` `pattern1_list_opt`
+             : `pattern1`
+   pattern1_list : `pattern1_list` `pattern1`
+                 : `pattern1`
+   pattern1_list_opt : `pattern1_list_opt` `pattern1`
+                     : `empty`
+   pattern1 : `pattern0` % `IDENT`
+            : `pattern0`
+   pattern0 : `reference`
+            : {| `record_patterns` |}
+            : _
+            : ( `pattern200` )
+            : ( `pattern200` : `lconstr` )
+            : `INT`
+            : `string`
+   impl_ident_tail : }
+                   : `name_list` : `lconstr` }
+                   : `name_list` }
+                   : : `lconstr` }
+   fixannot : { struct `ident` }
+            : { wf `constr` `ident_opt` }
+            : { measure `constr` `ident_opt` `constr_opt` }
+            : { struct `name` }
+            : `empty`
+   ident_opt : `ident`
+             : `empty`
+   constr_opt : `constr`
+              : `empty`
+   impl_name_head : `empty`
+   binders_fixannot : `impl_name_head` `impl_ident_tail` `binders_fixannot`
+                    : `fixannot`
+                    : `binder` `binders_fixannot`
+                    : `empty`
+   open_binders : `name` `name_list_opt2` : `lconstr`
+                : `name` `name_list_opt2` `binders`
+                : `name` .. `name`
+                : `closed_binder` `binders`
+   name_list_opt2 : `name_list_opt2` `name`
+                  : `empty`
+   binders : `binder_list_opt`
+   binder_list_opt : `binder_list_opt` `binder`
+                   : `empty`
+   binder : `name`
+          : `closed_binder`
+   typeclass_constraint : ! `operconstr200`
+                        : { `name` } : `exclam_opt` `operconstr200`
+                        : `name_colon` `exclam_opt` `operconstr200`
+                        : `operconstr200`
+   type_cstr : `lconstr_opt2`
+             : : `lconstr`
+             : `empty`
+   preident : `IDENT`
+   pattern_identref : ? `ident`
+   var : `ident`
+   field : `FIELD`
+   fields : `field` `fields`
+          : `field`
+   fullyqualid : `ident` `fields`
+               : `ident`
+   basequalid : `ident` `fields`
+              : `ident`
+   name : _
+        : `ident`
+   reference : `ident` `fields`
+             : `ident`
+   by_notation : `ne_string` `IDENT_opt`
+   IDENT_opt : % `IDENT`
+             : `empty`
+   smart_global : `reference`
+                : `by_notation`
+   qualid : `basequalid`
+   ne_string : `STRING`
+   ne_lstring : `ne_string`
+   dirpath : `ident` `field_list_opt`
+   field_list_opt : `field_list_opt` `field`
+                  : `empty`
+   string : `STRING`
+   lstring : `string`
+   integer : `INT`
+           : - `INT`
+   natural : `INT`
+   bigint : `INT`
```

Comment:
The `string` and `num` token are currently tokens that are defined in the text of the manual and `STRING` and `INT` could be mapped to them (assuming they can't be renamed in the `mlg` files).

Reply:
`STRING` and `INT` are tokens.  We could easily add nonterminals `string ::= STRING` and `int ::= INT` and then use the lowercase ones throughout the mlgs.

Also the name `INT` is a lie.  `INT` only includes natural numbers.